### PR TITLE
feat: make js-tracer feature optional for node builder crate

### LINKED ISSUE #11548: Make `js-tracer` optional in `rpc`
### Describe the feature

It's not possible to use `rpc` without having to compile all the `boa` stuff which takes a VERY long time to compile.

This was already reported in #10441, but partially implemented in #10444 for the `rpc-eth-types` and the `revm-inspectors`, but we're still missing in `rpc`: https://github.com/paradigmxyz/reth/blob/c35b8beb35594bb3192caec7d4335ca567d45539/crates/node/builder/Cargo.toml#L49

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by edisontim**
Closes #11548

### CODE REVIEW

File: crates/node/builder/Cargo.toml [hunk 1]
```diff
@@ -114,3 +114,4 @@ test-utils = [
 	"reth-provider/test-utils",
 	"reth-transaction-pool/test-utils"
 ]
+js-tracer = ["reth-rpc/js-tracer"]
```
Comment (mattsse) [inline crates/node/builder/Cargo.toml:117]: this should go above test-utils

**Review by mattsse — CHANGES_REQUESTED** [review]
need some clarification on the issue first to proceed here

**Review by mattsse — CHANGES_REQUESTED** [review]
we also need to enable this in the node crates for ethereum and optimism to actually enable this:

https://github.com/paradigmxyz/reth/blob/129f3ba9116da0fcbfafb78e66b7a0163e5464d3/crates/optimism/node/Cargo.toml#L23-L23

https://github.com/paradigmxyz/reth/blob/129f3ba9116da0fcbfafb78e66b7a0163e5464d3/crates/ethereum/node/Cargo.toml#L19-L19

**Comment by edisontim** [comment]
> we also need to enable this in the node crates for ethereum and optimism to actually enable this:
> 
> https://github.com/paradigmxyz/reth/blob/129f3ba9116da0fcbfafb78e66b7a0163e5464d3/crates/optimism/node/Cargo.toml#L23-L23
> 
> https://github.com/paradigmxyz/reth/blob/129f3ba9116da0fcbfafb78e66b7a0163e5464d3/crates/ethereum/node/Cargo.toml#L19-L19

Sure! Should I make this a default when building or also opt is as a feature?

**Comment by storopoli** [comment]
> > we also need to enable this in the node crates for ethereum and optimism to actually enable this:
> > https://github.com/paradigmxyz/reth/blob/129f3ba9116da0fcbfafb78e66b7a0163e5464d3/crates/optimism/node/Cargo.toml#L23-L23
> > 
> > https://github.com/paradigmxyz/reth/blob/129f3ba9116da0fcbfafb78e66b7a0163e5464d3/crates/ethereum/node/Cargo.toml#L19-L19
> 
> Sure! Should I make this a default or also make it as a feature in optimism and eth crate? `reth-node-builder = { workspace = true, features = ["js-tracer"] }` or
> 
> ```
> js-tracer = [
> 	"reth-node-builder/js-tracer"
> ]
> ```

If possible make it all a feature and opt-in.
The `boa` machinery should, in my humble opinion, be opt-in.
It really increases the compile time for users and also for everything that it encompasses such as CI for example.

**Comment by delbonis** [comment]
I'm also needing this.  This should be opt-in, at the very least because (as it is right now) it makes API consumer have to build the whole JS engine even when they don't use it.  I don't know if there's some issue with the crate hierarchy that makes the exposing the types in the API necessitate building the engine crates, that seems strange.

**Comment by edisontim** [comment]
Hey @mattsse, i think this is ready for review if you could have a look :)

**Comment by mattsse** [comment]
> I don't know if there's some issue with the crate hierarchy that makes the exposing the types in the API necessitate building the engine crates, that seems strange.

could please elaborate, which crates exactly @delbonis

**Review by mattsse — CHANGES_REQUESTED** [review]
this still removes js-tracer support from the binary because now it is no longer enabled.

we either need this for ethereum node and optimism node by default or must enable this in the binary.

**Comment by edisontim** [comment]
> this still removes js-tracer support from the binary because now it is no longer enabled.
> 
> we either need this for ethereum node and optimism node by default or must enable this in the binary.

Done !

File: bin/reth/Cargo.toml
Comment (mattsse) [inline bin/reth/Cargo.toml:1]: we need this for the optimimism/bin as well

**Review by mattsse — CHANGES_REQUESTED** [review]
we also n

**Comment by storopoli** [comment]
friendly bump 😄

**Comment by mattsse** [comment]
thanks, wrapping up here now

**Review by mattsse — APPROVED** [review]

**Comment by storopoli** [comment]
Nice thanks! :)
When does `reth` releases a new version?

**Pull request merged**